ui/pages/confirm-add-suggested-token/confirm-add-suggested-token.component.js

```diff
@@ -1,192 +1,49 @@
-              })}
-            </div>
-          </div>
-        </div>
-        <div className="page-container__footer">
-          <footer>
-            <Button
-              type="secondary"
-              large
-              className="page-container__footer-button"
-              onClick={async () => {
-                await Promise.all(
-                  suggestedAssets.map(async ({ id }) => rejectWatchAsset(id)),
-                );
-                history.push(mostRecentOverviewPage);
-              }}
-            >
-              {this.context.t('cancel')}
-            </Button>
-            <Button
-              type="primary"
-              large
-              className="page-container__footer-button"
-              disabled={suggestedAssets.length === 0}
-              onClick={async () => {
-                await Promise.all(
-                  suggestedAssets.map(async ({ asset, id }) => {
-                    await acceptWatchAsset(id);
-                    this.context.trackEvent({
-                      event: 'Token Added',
-                      category: 'Wallet',
-                      sensitiveProperties: {
-                        token_symbol: asset.symbol,
-                        token_contract_address: asset.address,
-                        token_decimal_precision: asset.decimals,
-                        unlisted: asset.unlisted,
-                        source: 'dapp',
-                      },
-                    });
-                  }),
-                );
-                history.push(mostRecentOverviewPage);
-              }}
-            >
-              {this.context.t('addToken')}
-            </Button>
-          </footer>
-        </div>
-      </div>
-    );
-  }
+  const hasTokenDuplicates = checkTokenDuplicates(suggestedAssets, tokens);
+  const reusesName = checkNameReuse(suggestedAssets, tokens);
 
-  checkTokenDuplicates(suggestedAssets, tokens) {
-    const pending = suggestedAssets.map(({ asset }) =>
-      asset.address.toUpperCase(),
-    );
-    const existing = tokens.map((token) => token.address.toUpperCase());
-    const dupes = pending.filter((proposed) => {
-      return existing.includes(proposed);
+  const tokenAddedEvent = (asset) => {
+    metricsEvent({
+      event: 'Token Added',
+      category: 'Wallet',
+      sensitiveProperties: {
+        token_symbol: asset.symbol,
+        token_contract_address: asset.address,
+        token_decimal_precision: asset.decimals,
+        unlisted: asset.unlisted,
+        source: 'dapp',
+      },
     });
-
-    return dupes.length > 0;
-  }
+  };
 
   /**
    * Returns true if any suggestedAssets both:
    * - Share a symbol with an existing `tokens` member.
    * - Does not share an address with that same `tokens` member.
    * This should be flagged as possibly deceptive or confusing.
-   *
-   * @param suggestedAssets
-   * @param tokens
    */
-  checkNameReuse(suggestedAssets, tokens) {
+  function checkNameReuse() {
```

Comment:
Could we use const arrow function declarations instead of `function` here to match existing patterns?

Reply:
Thank you! sounds good to me 

I also went with using an arrow function to declare the functional component since I have seen newer pushes to do it this way. I initially used `export default function` based on a 2020 PR https://github.com/MetaMask/metamask-extension/pull/7788

commit: a5a6815